File: csharp/examples/image_gallery/image_gallery.cs
using Uimd;

public sealed class ImageGallery : ImageGalleryUI
{
    private bool finished;

    protected override void onCloseBtnClick()
    {
        finished = true;
    }

    protected override bool shouldClose()
    {
        return finished;
    }
}

public static class Program
{
    public static int Main(string[] args)
    {
        return GeneratedWindowRuntime.RunGeneratedAppMain(() =>
        {
            ImageGallery app = new();
            return GeneratedWindowRuntime.RunGeneratedWindow(app, app.RuntimeOptions(), args);
        });
    }
}

File: swift/examples/image_gallery/image_gallery.swift
import Foundation
import Uimd

public final class ImageGallery: ImageGalleryUI
{
    private var finished = false

    override public func onCloseBtnClick()
    {
        finished = true
    }

    override public func shouldClose() -> Bool
    {
        finished
    }
}

@main
enum Program
{
    static func main()
    {
        let exitCode = GeneratedWindowRuntime.runGeneratedAppMain
        {
            let app = ImageGallery()
            return GeneratedWindowRuntime.runGeneratedWindow(app, app.runtimeOptions(), CommandLine.arguments)
        }
        Foundation.exit(Int32(exitCode))
    }
}
